collector/net.go

```diff
@@ -0,0 +1,161 @@
+// returns data points from Win32_PerfRawData_Tcpip_NetworkInterface
+
+// https://technet.microsoft.com/en-us/security/aa394340(v=vs.80) (Win32_PerfRawData_Tcpip_NetworkInterface class)
+
+package collector
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["net"] = NewNetworkCollector
+}
+
+var (
+	nicWhitelist = flag.String("collector.net.nic-whitelist", ".+", "Regexp of NIC:s to whitelist. NIC name must both match whitelist and not match blacklist to be included.")
+	nicBlacklist = flag.String("collector.net.nic-blacklist", "", "Regexp of NIC:s to blacklist. NIC name must both match whitelist and not match blacklist to be included.")
+)
+
+// A NetworkCollector is a Prometheus collector for WMI Win32_PerfRawData_Tcpip_NetworkInterface metrics
+type NetworkCollector struct {
+	BytesReceivedTotal   *prometheus.Desc
+	BytesSentTotal       *prometheus.Desc
+	BytesTotal           *prometheus.Desc
+	PacketsTotal         *prometheus.Desc
+	PacketsReceivedTotal *prometheus.Desc
+	PacketsSentTotal     *prometheus.Desc
+
+	nicWhitelistPattern *regexp.Regexp
+	nicBlacklistPattern *regexp.Regexp
+}
+
+// NewNetworkCollector ...
+func NewNetworkCollector() (Collector, error) {
+	const subsystem = "net"
+
+	return &NetworkCollector{
+		BytesReceivedTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "bytes_received_total"),
+			"(Network.BytesReceivedPerSec)",
+			[]string{"nic"},
```

Comment:
On my local machine, I get this `Intel[R] Ethernet Connection I218-LM` - could this kind of labels make querying hard? Not sure what to do about it if so, but we might want to think about it.